rotkehlchen/tests/external_apis/test_coingecko.py

```diff
@@ -63,3 +68,64 @@ def test_assets_with_icons(icon_manager):
     """Checks that _assets_with_coingecko_id returns a proper result"""
     x = icon_manager._assets_with_coingecko_id()
     assert len(x) > 1000
+
+
+def test_asset_icons_for_collections(icon_manager: IconManager) -> None:
+    """
+    Test that for assets in the same collection only one file is saved and
+    is later used by all the assets in the same collection
+    """
+    dai_op = Asset('eip155:10/erc20:0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1')
+    times_api_was_queried = 0
+
+    # remove icon file in case it was added by another test
+    icon_path = icon_manager.iconfile_path(A_DAI)
+    icon_path.unlink(missing_ok=True)
+    assert icon_path.exists() is False
+
+    # mock coingecko response
+    def mock_coingecko(url, timeout):  # pylint: disable=unused-argument
+        nonlocal times_api_was_queried
+        times_api_was_queried += 1
+        test_data_folder = Path(__file__).resolve().parent.parent / 'data' / 'mocks' / 'test_coingecko'  # noqa: E501
+        if 'https://api.coingecko.com/api/v3/coins/dai' in url:
+            with open(test_data_folder / 'coins' / 'dai.json') as f:
+                return MockResponse(HTTPStatus.OK, f.read())
+        elif 'https://api.coingecko.com/api/v3/coins/ethereum' in url:
+            with open(test_data_folder / 'coins' / 'ethereum.json') as f:
+                return MockResponse(HTTPStatus.OK, f.read())
+        elif url in {
+            'https://assets.coingecko.com/coins/images/9956/small/4943.png?1636636734',
+            'https://assets.coingecko.com/coins/images/279/small/ethereum.png?1595348880',
+        }:
+            icon_name = '4943.png'
+            if '279' in url:
+                icon_name = '279.png'
+            with open(test_data_folder / 'icons' / icon_name, 'rb') as f:
+                return MockResponse(
+                    status_code=HTTPStatus.OK,
+                    text=str(f.read()),
+                    headers={'Content-Type': 'image/png'},
+                )
+
+        raise AssertionError(f'Unexpected url {url} in bitpanda test')
```

Comment:
I can see where you copied the logic from

```suggestion
        raise AssertionError(f'Unexpected url {url} in asset collection icons test)
```